test/lib/ufe/testAttributeBlock.py

```diff
@@ -174,5 +174,92 @@ def testTransformationAttributeBlocking(self):
+        subLayerC = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerC")[0]
+        subLayerB = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerB")[0]
+        subLayerA = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerA")[0]
+
+        # check to see the sublayers added
+        addedLayers = [subLayerA, subLayerB, subLayerC]
+        self.assertEqual(rootLayer.subLayerPaths, addedLayers)
+
+        # set the edit target to LayerB
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerB)
+
+        # create a xformable and give it a common API single "pivot".
+        # This will match the common API, but not the Maya API.
+        cylinderXformable = UsdGeom.Xformable(cylinderPrim)
+
+        cylinderXformable.AddTranslateOp(UsdGeom.XformOp.PrecisionFloat, "pivot")
+        cylinderXformable.AddTranslateOp(
+            UsdGeom.XformOp.PrecisionFloat, "pivot", True)
+
+        self.assertEqual(
+            cylinderXformable.GetXformOpOrderAttr().Get(), 
+            Vt.TokenArray(("xformOp:translate:pivot",
+                           "!invert!xformOp:translate:pivot")))
+        self.assertTrue(UsdGeom.XformCommonAPI(cylinderXformable))
+
+        # Now that we have set up the transform stack to be common API-compatible,
+        # create the Transform3d interface.
+        cylinderPath = ufe.Path([
+                     mayaUtils.createUfePathSegment("|mayaUsdTransform|shape"), 
+                     usdUtils.createUfePathSegment("/pCylinder1")])
+        cylinderItem = ufe.Hierarchy.createItem(cylinderPath)
+
+        cylinderT3d = ufe.Transform3d.transform3d(cylinderItem)
+
+        # select the cylinderItem
+        sn = ufe.GlobalSelection.get()
+        sn.clear()
+        sn.append(cylinderItem)
+
+        # rotate the cylinder
+        cmds.rotate(0, 90, 0, relative=True, objectSpace=True, forceOrderXYZ=True)
+
+        self.assertEqual(
+            cylinderXformable.GetXformOpOrderAttr().Get(), 
+            Vt.TokenArray(("xformOp:translate:pivot", "xformOp:rotateXYZ",
+                           "!invert!xformOp:translate:pivot")))
+        self.assertTrue(UsdGeom.XformCommonAPI(cylinderXformable))
+
+        # set the edit target to a weaker layer (LayerC)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerC)
+
+        # check if rotate attribute is editable
+        rotateAttr = cylinderPrim.GetAttribute('xformOp:rotateXYZ')
+        self.assertIsNotNone(rotateAttr)
+
+        # authoring new transformation edit is not allowed.
+        self.assertFalse(mayaUsdUfe.isAttributeEditAllowed(rotateAttr))
+
+        # set the edit target to a stronger layer (LayerA)
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerA)
+
+        # authoring new transformation edit is allowed.
+        self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(rotateAttr))
```

Comment:
Unit test to cover attribute blocking using UsdGeomXformCommonAPI. 

@ppt-adsk I don't have a test coverage for UsdTransform3dMatrixOp just yet. I tried setting up a simple transform op with an identity matrix ( Matrix4d ) but I wasn't able to call into UsdTransform3dMatrixOp. Perhaps I am missing something simple here. 

We also don't have any test coverage for UsdTransform3dMatrixOp. Based on the conversation we had the other day, Houdini Transform SOP falls into Transform3dMatrixOp. I am setting up an example to see how it works.